### PR TITLE
Destroy virtual thread data after VirtualThreadEnd callback

### PR DESCRIPTION
**Pull request by thallium**
Currently destroyThreadData() is hooked onto the VirtualThreadEnd event which is called before the event callback. In JDWP, the agent uses TLS to associate nodes in the running thread list. Clearing TLS before the event callback makes the agent lose track of the node thus failed to remove existed virtual threads from the list. This commit adds JVMTI VirtualThreadDestroy event which is triggered after VirtualThreadEnd and jvmtiHookVirtualThreadDestroy() is hooked onto VirtualThreadDestroy event.

Fixes:
https://github.com/ibmruntimes/Semeru-Runtimes/issues/80 
https://github.com/eclipse-openj9/openj9/issues/19759

### CODE REVIEW

**Comment by thallium** [comment]
@babsingh requesting your review

**Comment by thallium** [comment]
@babsingh I've updated the PR and can you please review again?

**Review by babsingh — COMMENTED** [review]
- PR's description should be updated to reflect the latest commit message.
- Was a personal build with the `opendk.extended` test target run?

**Comment by thallium** [comment]
Will run `opendk.extended` tests and post the result.

**Comment by babsingh** [comment]
> Will run openjdk.extended tests and post the result.

Changes LGTM. I will launch it as part of PR builds.

**Comment by babsingh** [comment]
Jenkins test sanity,openjdk.sanity,openjdk.extended zlinux jdk21

**Comment by babsingh** [comment]
Jenkins test sanity,openjdk.sanity,openjdk.extended amac jdk23

**Comment by babsingh** [comment]
Jenkins compile win jdk23

**Comment by babsingh** [comment]
Jenkins test sanity,sanity.openjdk,extended.openjdk zlinux jdk21

**Comment by babsingh** [comment]
Jenkins test sanity,sanity.openjdk,extended.openjdk amac jdk23

**Comment by babsingh** [comment]
Jenkins test sanity,sanity.openjdk,extended.openjdk amac jdk21

**Comment by babsingh** [comment]
Jenkins compile win jdk21

**Comment by babsingh** [comment]
@thallium Create a PR to push these changes to the [v0.47.0-release](https://github.com/eclipse-openj9/openj9/tree/v0.47.0-release) branch.

File: runtime/nls/j9ti/jvmti.nls [hunk 1]
```diff
@@ -639,6 +639,13 @@ J9NLS_JVMTI_COM_SUN_HOTSPOT_EVENTS_VIRTUAL_THREAD_UNMOUNT.system_action=None
 J9NLS_JVMTI_COM_SUN_HOTSPOT_EVENTS_VIRTUAL_THREAD_UNMOUNT.user_response=None
 # END NON-TRANSLATABLE
 
+J9NLS_JVMTI_COM_SUN_HOTSPOT_EVENTS_VIRTUAL_THREAD_DESTROY=Free native resources after a virtual thread ends.
+# START NON-TRANSLATABLE
+J9NLS_JVMTI_COM_SUN_HOTSPOT_EVENTS_VIRTUAL_THREAD_DESTROY.explanation=Internationalized description of a JVMTI extension
+J9NLS_JVMTI_COM_SUN_HOTSPOT_EVENTS_VIRTUAL_THREAD_DESTROY.system_action=None
+J9NLS_JVMTI_COM_SUN_HOTSPOT_EVENTS_VIRTUAL_THREAD_DESTROY.user_response=None
+# END NON-TRANSLATABLE
```
Comment (keithc-ca) [inline runtime/nls/j9ti/jvmti.nls:647]: New NLS messages must be added at the end of the file (see line 27).

Comment (keithc-ca) [reply runtime/nls/j9ti/jvmti.nls:647]: See https://github.com/eclipse-openj9/openj9/pull/19873.

**Pull request merged**